commit dc66f7b6d0e02f09e08d80042f211a6336e7d8ff
Author: Pierre Rineau <pierre.rineau@makina-corpus.com>
Date:   Thu Dec 1 13:55:43 2016 +0100

    Issue #2832083: ThemeRegistryLoader::findTemplate() breaks Twig_Loader_Filesystem signature

diff --git a/core/lib/Drupal/Core/Template/Loader/ThemeRegistryLoader.php b/core/lib/Drupal/Core/Template/Loader/ThemeRegistryLoader.php
index 67a82dd..2c6aa3e 100644
--- a/core/lib/Drupal/Core/Template/Loader/ThemeRegistryLoader.php
+++ b/core/lib/Drupal/Core/Template/Loader/ThemeRegistryLoader.php
@@ -63,6 +63,8 @@ protected function findTemplate($name, $throw = TRUE) {
     if ($throw) {
       throw new \Twig_Error_Loader(sprintf('Unable to find template "%s" in the Drupal theme registry.', $name));
     }
+
+    return FALSE;
   }
 
 }
